### PR TITLE
Drag&drop textures from filesystem to visual shader

### LINKED ISSUE #21662: Drag and drop image to visual shader should create constant texture node immediately
**Godot version:**

3.1 alpha

**OS/device including version:**

nil

**Issue description:**

I am playing with the visual shader editor and now I can drag and drop the image into a node after I create the constant texture node, it will be more direct if one can just drag and drop the image into the editor and the node will be created automatically. As in most case, I believe that's how a texture will be used in a shader.

**Steps to reproduce:**

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by Chaosus**
I guess, this is desired functionality, Closes #21662

draganddrop

### CODE REVIEW

**Comment by lentsius-bark** [comment]
Oh my. My jaw drops with your constant goodness you've been adding to the visual shader! Thank you!

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 1]
```diff
@@ -2054,6 +2068,28 @@ void VisualShaderEditor::drop_data_fw(const Point2 &p_point, const Variant &p_da
...
 			_add_node(idx, add_options[idx].sub_func);
+		} else if (d.has("files")) {
+			if (d["files"].get_type() == Variant::POOL_STRING_ARRAY) {
+
+				PoolStringArray arr = d["files"];
+				for (int i = 0; i < arr.size(); i++) {
+
+					String type = ResourceLoader::get_resource_type(arr[i]);
+
+					if (type == "Texture" ||
```
Comment (akien-mga) [inline editor/plugins/visual_shader_editor_plugin.cpp]: On phone right now so can't check, but I think there might be a ClassDB method to check if the type inheritd Texture.

Comment (Chaosus) [reply editor/plugins/visual_shader_editor_plugin.cpp]: Fixed, also improved performance by replacing loop to index insertion in _add_texture_node

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**